gateway/src/apicast/policy/rate_limit/apicast-policy.json

```diff
@@ -11,29 +11,54 @@
         "description": "List of limiters to be applied",
         "type": "array",
         "items": {
-          "oneOf": [{
+          "anyOf": [{
             "type": "object",
             "properties": {
               "name": {
                 "type": "string",
                 "enum": ["connections"],
-                "description": "limiting request concurrency (or concurrent connections)"
+                "description": "Limiting request concurrency (or concurrent connections)"
               },
               "key": {
-                "description": "Key of limiter",
-                "type": "string"
+                "description": "The key corresponding to the limiter object",
+                "type": "object",
+                "properties": {
+                  "name": {
+                    "type": "string",
+                    "description": "The name of the key, must be unique in the scope"
+                  },
+                  "scope": {
+                    "type": "string",
+                    "description": "Scope of the key",
+                    "default": "global",
+                    "oneOf": [{
+                      "enum": ["global"],
+                      "description": "Global scope, affecting to all services"
+                    }, {
+                      "enum": ["service"],
+                      "description": "Service scope, affecting to one service"
+                    }]
+                  },
+                  "service_name": {
```

Comment:
I thought this won't be supplied by users but rather just automatically using the service_id available in the context. So all service scope limiters are automatically scoped and can't be shared between services. This could still be shared by multiple services when they fill the same identifier.